bookkeeper-server/pom.xml

```diff
@@ -202,19 +202,16 @@
         </configuration>
         <executions>
           <execution>
+            <id>create-shaded-jar</id>
             <phase>package</phase>
             <goals>
               <goal>shade</goal>
             </goals>
             <configuration>
               <createDependencyReducedPom>true</createDependencyReducedPom>
-              <promoteTransitiveDependencies>true</promoteTransitiveDependencies>
```

Comment:
with these lines, the shaded jar will not work?

Reply:
seems we should also config the <relocations> also?

Reply:
@eolivelli have you tried this before?

Reply:
No, but actually we don't need to change the package name for our own module.

The problem is about the dependencies declared in the pom if the  shaded artifact.
It seems that the plugin does not publish the reduced pom but the original one

Reply:
No, but actually we don't need to change the package name for our own module.

The problem is about the dependencies declared in the pom if the  shaded artifact.
It seems that the plugin does not publish the reduced pom but the original one

Reply:
It is the createDependencyReducedPom that it is not working.
Maybe it is a bug in the plugin. Le'ts see @sijie findings.

I think the base jar should be non-shaded but we should also provide the jar without original Guava and Protobuf. These jar are evil in the Hadoop ecosystem look for "Hbase guava" or "hadoop guava", similar is protobuf

Reply:
I see, It is createDependencyReducedPom that is required.
Seems other guy have met the same [problem](http://maven.40175.n5.nabble.com/shade-plugin-doesn-t-create-dependency-reduced-pom-td4331135.html).  

In my test, after turn `shadedArtifactAttached` into false, it will create a `./bookkeeper-server/dependency-reduced-pom.xml`, but this change will meet another error
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-shade-plugin:2.4.3:shade (default) on project bookkeeper-server: Error creating shaded jar: Could not resolve following dependencies: [javax.jms:jms:jar:1.1 (compile), com.sun.jdmk:jmxtools:jar:1.2.1 (compile), com.sun.jmx:jmxri:jar:1.2.1 (compile)]: Could not resolve dependencies for project org.apache.bookkeeper:bookkeeper-server:jar:4.7.0-SNAPSHOT: The following artifacts could not be resolved: javax.jms:jms:jar:1.1, com.sun.jdmk:jmxtools:jar:1.2.1, com.sun.jmx:jmxri:jar:1.2.1: Could not transfer artifact javax.jms:jms:jar:1.1 from/to java.net (https://maven-repository.dev.java.net/nonav/repository): Cannot access https://maven-repository.dev.java.net/nonav/repository with type legacy using the available connector factories: BasicRepositoryConnectorFactory: Cannot access https://maven-repository.dev.java.net/nonav/repository with type legacy using the available layout factories: Maven2RepositoryLayoutFactory: Unsupported repository layout legacy -> [Help 1]
```